parachain-tracer/src/tracker.rs

```diff
@@ -15,901 +15,1156 @@
+				matches!(rpc.occupied_cores(block_hash).await?[core as usize], CoreOccupied::Paras);
+			self.is_on_demand_scheduled_in_current_block =
+				self.on_demand_order.is_some() && scheduled_ids[0] == self.para_id;
 		}
-
 		Ok(())
 	}
 
-	fn update_backing(
-		&mut self,
-		mut backed_candidates: Vec<polkadot_primitives::BackedCandidate<H256>>,
-		block_number: BlockNumber,
-	) -> bool {
-		let candidate_index = backed_candidates
-			.iter()
-			.position(|candidate| candidate.candidate.descriptor.para_id.0 == self.para_id);
-
-		// Update the curent state if a candiate was backed for this para.
-		if let Some(index) = candidate_index {
-			self.current_candidate.state = ParachainBlockState::Backed;
-			let backed_candidate = backed_candidates.remove(index);
-			let commitments_hash = BlakeTwo256::hash_of(&backed_candidate.candidate.commitments);
-			self.current_candidate.candidate_hash =
-				Some(BlakeTwo256::hash_of(&(&backed_candidate.candidate.descriptor, commitments_hash)));
-			self.current_candidate.candidate = Some(backed_candidate);
-			self.last_backed_at = Some(block_number);
-
-			true
-		} else {
-			false
+	async fn set_disputes(&mut self, disputes: &[DisputeStatementSet], storage: &TrackerStorage) {
+		self.disputes = Vec::with_capacity(disputes.len());
+		for dispute_info in disputes {
+			if let Some(DisputeInfo { outcome, session_index, initiated, initiator_indices, concluded, .. }) =
+				storage.dispute(dispute_info.candidate_hash.0).await
+			{
+				if let Some(outcome) = outcome {
+					self.disputes.push(DisputesTracker::new(
+						dispute_info,
+						outcome,
+						initiated,
+						initiator_indices,
+						concluded.expect("dispute must be concluded"),
+						storage.session_keys(dispute_info.session).await.as_ref(),
+						storage.session_keys(session_index).await.as_ref(),
+					));
+				} else {
+					info!(
+						"parachain {}: dispute for candidate {} has been seen in the block inherent but is not tracked to be resolved",
+						self.para_id,
+						dispute_info.candidate_hash.0
+					);
+				}
+			}
 		}
 	}
 
-	async fn get_core_assignments_via_scheduled_paras(
+	async fn set_availability(
 		&mut self,
 		block_hash: H256,
-	) -> color_eyre::Result<HashMap<u32, Vec<u32>>, SubxtWrapperError> {
-		let core_assignments = self
-			.executor
-			.get_scheduled_paras(self.node_rpc_url.as_str(), block_hash)
-			.await?;
+		bitfields: Vec<AvailabilityBitfield>,
+		rpc: &mut impl TrackerRpc,
+	) -> color_eyre::Result<()> {
+		if self.current_candidate.is_backed() {
+			// We only process availability if our parachain is assigned to an availability core.
+			if let Some(core) = self.current_candidate.assigned_core {
+				self.current_candidate.current_availability_bits = extract_availability_bits_count(bitfields, core);
+				self.current_candidate.max_availability_bits =
+					self.validators_indices(block_hash, rpc).await?.len() as u32;
+
+				if self.current_candidate.is_data_available() {
```

Comment:
These are the availability bits for a previous candidate. We need to account for async backing.